calico_containers/calico_ctl/profile.py

```diff
@@ -150,14 +153,35 @@ def profile_add(profile_name):
         print "Created profile %s" % profile_name
 
 
-def profile_remove(profile_name):
-    # TODO - Don't allow removing a profile that has endpoints in it.
-    try:
-        client.remove_profile(profile_name)
-    except KeyError:
-        print "Couldn't find profile with name %s" % profile_name
+def profile_remove(profile_name, nocheck):
+    """
+    Remove a profile as long as it does not contain any endpoints.
+    Allow user to explicitly remove the profile if desired.
+    :param profile_name: The name of the profile to remove.
+    :param nocheck: Flag saying to remove profile regardless of endpoints.
+    :return: None.
+    """
+    # Check if the profile exists.
+    if client.profile_exists(profile_name):
+        rm_profile = False
+        # Check that the nocheck flag was used
+        if nocheck:
+            rm_profile = True
+        else:
+            # Check if the the profile has endpoints associated with it
+            members = client.get_profile_members(profile_name)
+            if not members:
+                rm_profile = True
+        # Remove the profile if criteria was met
+        if rm_profile:
+            client.remove_profile(profile_name)
+            print "Deleted profile %s" % profile_name
+        else:
+            # Members must exist if this branch is reached
+            print "Cannot remove profile - profile contains endpoints.\n" + \
```

Comment:
Ah, just realised - not sure about this wording.  Maybe run by Tom too...

IMHO, a profile doesn't really contain endpoints, a profile is referenced by one or more endpoints.